### PR TITLE
Improvement: Better Alignment + Outline Improvement in Custom Scoreboard

### PR DESCRIPTION
**Pull request by j10a1n15**
## What
This Pull Request adds more alignment options (Left/Center; Top/Center) to the Custom Scoreboard Alignment options.

## Changelog Improvements
+ Added more Alignment Options to Custom Scoreboard. - j10a1n15
+ Allowed Custom Scoreboard Outline to also work on Background Images. - j10a1n15

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/config/ConfigUpdaterMigrator.kt [hunk 1]
```diff
@@ -12,7 +12,7 @@ import com.google.gson.JsonPrimitive
 object ConfigUpdaterMigrator {
 
     val logger = LorenzLogger("ConfigMigration")
-    const val CONFIG_VERSION = 41
+    const val CONFIG_VERSION = 42
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/ConfigUpdaterMigrator.kt]: this needs to be 43 now

File: src/main/java/at/hannibal2/skyhanni/config/features/gui/customscoreboard/AlignmentConfig.java [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
 public class AlignmentConfig {
-    // TODO: Switch to Dropdowns with multiple different alignment ways in the future
-    // Horizontal: Left, Center, Right
-    // Vertical: Top, Center, Bottom
+
     @Expose
-    @ConfigOption(name = "Align to the right", desc = "Align the scoreboard to the right side of the screen.")
-    @ConfigEditorBoolean
-    public boolean alignRight = true;
+    @ConfigOption(name = "Align Horizontally", desc = "Align the scoreboard on the horizontal axis.")
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/gui/customscoreboard/AlignmentConfig.java]:
```suggestion
    @ConfigOption(name = "Horizontal Alignment", desc = "Alignment for the scoreboard on the horizontal axis.")
```

File: src/main/java/at/hannibal2/skyhanni/config/features/gui/customscoreboard/AlignmentConfig.java [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
+    @ConfigEditorDropdown
+    public RenderUtils.HorizontalAlignment horizontalAlignment = RenderUtils.HorizontalAlignment.RIGHT;
 
     @Expose
-    @ConfigOption(name = "Align to the center vertically", desc = "Align the scoreboard to the center of the screen vertically.")
-    @ConfigEditorBoolean
-    public boolean alignCenterVertically = true;
+    @ConfigOption(name = "Align Vertically", desc = "Align the scoreboard on the vertical axis.")
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/gui/customscoreboard/AlignmentConfig.java]:
```suggestion
    @ConfigOption(name = "Vertical Alignment", desc = "Alignment for the scoreboard on the vertical axis."
```

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 4]
```diff
@@ -231,5 +234,19 @@ class CustomScoreboard {
             newArray
         }
 
+        event.move(42, "$displayPrefix.alignment.alignRight", "$displayPrefix.alignment.horizontalAlignment") {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]:
```suggestion
        event.move(43, "$displayPrefix.alignment.alignRight", "$displayPrefix.alignment.horizontalAlignment") {
```

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 5]
```diff
@@ -231,5 +234,19 @@ class CustomScoreboard {
...
+            if (it.asBoolean) {
+                JsonPrimitive(HorizontalAlignment.RIGHT.name)
+            } else {
+                JsonPrimitive(HorizontalAlignment.DONT_ALIGN.name)
+            }
+        }
+        event.move(42, "$displayPrefix.alignment.alignCenterVertically", "$displayPrefix.alignment.verticalAlignment") {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]:
```suggestion
        event.move(43, "$displayPrefix.alignment.alignCenterVertically", "$displayPrefix.alignment.verticalAlignment") {
```

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
few small things

File: src/main/java/at/hannibal2/skyhanni/config/ConfigUpdaterMigrator.kt (hunk 1 above)
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/config/ConfigUpdaterMigrator.kt]:
```suggestion
    const val CONFIG_VERSION = 43
```

**Review by CalMWolfs — APPROVED** [review]
code looks good, havent tested in game

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 6]
```diff
@@ -231,5 +234,19 @@ class CustomScoreboard {
             newArray
         }
 
+        event.move(43, "$displayPrefix.alignment.alignRight", "$displayPrefix.alignment.horizontalAlignment") {
+            if (it.asBoolean) {
+                JsonPrimitive(HorizontalAlignment.RIGHT.name)
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]: move `JsonPrimitive(` outside the if. same below

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 7]
```diff
@@ -231,5 +234,19 @@ class CustomScoreboard {
...
+            } else {
+                JsonPrimitive(HorizontalAlignment.DONT_ALIGN.name)
+            }
+        }
+        event.move(43, "$displayPrefix.alignment.alignCenterVertically", "$displayPrefix.alignment.verticalAlignment") {
+            if (it.asBoolean) {
+                JsonPrimitive(RenderUtils.VerticalAlignment.CENTER.name)
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]: make RenderUtils import consistent (see above)

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 8]
```diff
@@ -68,9 +69,11 @@ class CustomScoreboard {
     @SubscribeEvent
     fun onGuiPositionMoved(event: GuiPositionMovedEvent) {
         if (event.guiName == guiName) {
-            if (alignmentConfig.alignRight || alignmentConfig.alignCenterVertically) {
-                alignmentConfig.alignRight = false
-                alignmentConfig.alignCenterVertically = false
+            if (alignmentConfig.horizontalAlignment != HorizontalAlignment.DONT_ALIGN
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]: maybe use `with(alignmentConfig) {` here

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/RenderBackground.kt [hunk 9]
```diff
@@ -33,14 +33,36 @@ class RenderBackground {
 
         // Update the position to the alignment options
         if (
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/RenderBackground.kt:33]: please move the whole if block into a new function for readabilty

File: src/main/java/at/hannibal2/skyhanni/utils/RenderUtils.kt [hunk 10]
```diff
@@ -47,17 +47,20 @@ object RenderUtils {
         LEFT("Left"),
         CENTER("Center"),
         RIGHT("Right"),
+        DONT_ALIGN("Don't Align")
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/utils/RenderUtils.kt]: missing comma

**Review by hannibal002 — CHANGES_REQUESTED** [review]
just code cleanup stuff. have not tested yet

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**